ratpack-session/src/main/java/ratpack/session/SessionCookieConfig.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ratpack.session;
+
+import java.time.Duration;
+
+/**
+ * Basic configuration for cookies related to session management.
+ * <p>
+ * Attributes are shared with {@link ratpack.session.clientside.ClientSideSessionConfig} that is used by client side session (cookie based).
+ *
+ * @see ratpack.session.clientside.ClientSideSessionConfig
+ */
+public class SessionCookieConfig {
+
+  private Duration expires;
+  private String domain;
+  private String path = "/";
+  private String name = "JSESSIONID";
+  private boolean httpOnly = true;
+  private boolean secure;
+
+  /**
+   * Cookie's max age.
+   *
+   * @return the max age of the session related cookies
+   */
+  public Duration getExpires() {
+    return expires;
+  }
+
+  /**
+   * Use the session cookie only when requesting from the {@code domain}.
+   * <p>
+   * Define the scope for the cookie.
+   *
+   * @return the URI domain to which session cookie will be attached to.
+   */
+  public String getDomain() {
+    return domain;
+  }
+
+  /**
+   * Use the session cookie only when requesting from the {@code path}.
+   * <p>
+   * Define the scope of the cookie.
+   * <p>
+   * Session should be send for every request. The {@code path} of value {@code "/"} does this.
+   * @return the URI path to which session cookie will be attached to.
+   */
+  public String getPath() {
+    return path;
+  }
+
+  /**
+   * The name of the cookie for session id.
+   * <p>
+   * Defaults to: {@code JSESSIONID}
+   *
+   * @return the name of the cookie for session id
+   */
+  public String getName() {
```

Comment:
Need to rename this to make it clear it's for the session id cookie name.

`getIdName()` ?